x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/action/TransportStartDataFrameTransformAction.java

```diff
@@ -133,7 +132,26 @@ protected void masterOperation(StartDataFrameTransformAction.Request request,
                             "Unable to start data frame transform [" + config.getId() +
                                 "] as it is in state [" + transformState.getTaskState()  + "]", RestStatus.CONFLICT));
                     } else {
-                        persistentTaskActionListener.onResponse(existingTask);
+                        // If the task already exists but is not assigned to a node, something is weird
+                        // return a failure that includes the current assignment explanation (if one exists)
+                        if (existingTask.isAssigned() == false) {
+                            String assignmentExplanation = "";
+                            if (existingTask.getAssignment() != null) {
+                                assignmentExplanation = existingTask.getAssignment().getExplanation();
+                            }
+                            listener.onFailure(new ElasticsearchStatusException("Unable to start data frame transform [" +
+                                config.getId() + "] as it is not assigned to a node, explanation" + assignmentExplanation,
+                                RestStatus.CONFLICT));
+                            return;
+                        }
+                        // If the task already exists and is assigned to a node, simply attempt to set it to start
+                        ClientHelper.executeAsyncWithOrigin(client,
```

Comment:
If the task exists and is assigned to a node then that means it is stopped, starting or started. I thought we had agreed on a change to remove the P task when the DF has stopped in which case this is not needed.

Reply:
Or @davidkyle  it is in a `FAILED` state.